patchwork-extensions-block/src/main/java/net/patchworkmc/mixin/extensions/block/harvest/MixinServerPlayerInteractionManager.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Minecraft Forge, Patchwork Project
+ * Copyright (c) 2016-2020, 2019-2020
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.patchworkmc.mixin.extensions.block.harvest;
+
+import javax.annotation.Nullable;
+
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Unique;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Redirect;
+import net.minecraftforge.common.extensions.IForgeBlockState;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockState;
+import net.minecraft.block.entity.BlockEntity;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.item.ItemStack;
+import net.minecraft.server.network.ServerPlayerEntity;
+import net.minecraft.server.network.ServerPlayerInteractionManager;
+import net.minecraft.server.world.ServerWorld;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.IWorld;
+import net.minecraft.world.World;
+
+import net.patchworkmc.impl.extensions.block.BlockContext;
+import net.patchworkmc.impl.extensions.block.BlockHarvestManager;
+import net.patchworkmc.impl.extensions.block.Signatures;
+
+@Mixin(ServerPlayerInteractionManager.class)
+public abstract class MixinServerPlayerInteractionManager {
+	// removedByPlayer, canHarvestBlock
+
+	@Unique
+	private boolean patchwork$removeBlock(BlockPos pos, boolean canHarvest) {
+		ServerPlayerInteractionManager me = (ServerPlayerInteractionManager) (Object) this;
+		BlockState state = me.world.getBlockState(pos);
+		boolean removed = ((IForgeBlockState) state).removedByPlayer(me.world, pos, me.player, canHarvest, me.world.getFluidState(pos));
+
+		if (removed) {
+			state.getBlock().onBroken(me.world, pos, state);
+		}
+
+		return removed;
+	}
+
+	@Redirect(method = "tryBreakBlock", at = @At(value = "INVOKE", target = Signatures.Block_onBreak, ordinal = 0))
+	private void patchwork$tryBreakBlock_onBreak(Block block, World world, BlockPos pos, BlockState state, PlayerEntity player) {
+		// Suppress this call
+	}
+
+	@Redirect(method = "tryBreakBlock", at = @At(value = "INVOKE", target = Signatures.ServerWorld_removeBlock, ordinal = 0))
```

Comment:
This is a bit weird, because I'm having trouble finding a "removeBlock" on ServerWorld, let along pointing out where it's called in the yarnforge patch. My best guess is what's called clearBlockState in yarnforge, but I can't find that on ServerWorld either :/

Don't worry about it too much, it's probably something weird in my setup.

Reply:
removeBlock is implemented in World not ServerWorld, I believe

Reply:
Yep, you're right.